Sources/MessagingInApp/MessagingInAppImplementation.swift

```diff
@@ -5,114 +5,62 @@ class MessagingInAppImplementation: MessagingInAppInstance {
     private let moduleConfig: MessagingInAppConfigOptions
 
     private let logger: Logger
-    private var inAppProvider: InAppProvider
-
-    private var eventListener: InAppEventListener?
+    private let inAppMessageManager: InAppMessageManager
+    private let gist: Gist
     private let threadUtil: ThreadUtil
-    let eventBusHandler: EventBusHandler
+    private let eventBusHandler: EventBusHandler
 
     init(diGraph: DIGraphShared, moduleConfig: MessagingInAppConfigOptions) {
         self.moduleConfig = moduleConfig
         self.logger = diGraph.logger
-        self.inAppProvider = diGraph.inAppProvider
+        self.inAppMessageManager = diGraph.inAppMessageManager
+        self.gist = Gist.shared
         self.threadUtil = diGraph.threadUtil
         self.eventBusHandler = diGraph.eventBusHandler
-        initialize()
+
+        subscribeToInAppMessageState()
     }
 
-    private func initialize() {
-        inAppProvider.initialize(
+    private func subscribeToInAppMessageState() {
+        inAppMessageManager.dispatch(action: .initialize(
             siteId: moduleConfig.siteId,
-            region: moduleConfig.region,
-            delegate: self,
-            enableLogging: logger.logLevel == .debug
-        )
+            dataCenter: moduleConfig.region.rawValue,
+            environment: GistEnvironment.production
+        )) {
+            self.subscribeToEventBus()
+        }
+    }
 
+    private func subscribeToEventBus() {
         // if identifier is already present, set the userToken again so in case if the customer was already identified and
         // module was added later on, we can notify gist about it.
         eventBusHandler.addObserver(ProfileIdentifiedEvent.self) { event in
             self.logger.debug("registering profile \(event.identifier) for in-app")
 
-            self.inAppProvider.setProfileIdentifier(event.identifier)
+            self.gist.setUserToken(event.identifier)
         }
 
         eventBusHandler.addObserver(ScreenViewedEvent.self) { event in
             self.logger.debug("setting route for in-app to \(event.name)")
 
             // Gist expects webview to be launched in main thread and changing route will trigger locally stored in-app messages for that route.
-            self.threadUtil.runMain {
```

Comment:
think this is because this is going to be done in middleware? do we still need the comment above?

Reply:
Nope. Removed comment